partition/bootloader.go

```diff
@@ -127,10 +127,15 @@ type bootloaderType struct {
 }
 
 func newBootLoader(partition *Partition, bootloaderDir string) *bootloaderType {
-	// FIXME: is this the right thing to do? i.e. what should we do
-	//        on a single partition system?
-	if partition.otherRootPartition() == nil {
-		return nil
+	// no root partition means we are in an all-snap system
+	if partition.rootPartition() == nil || partition.otherRootPartition() == nil {
```

Comment:
The partition/ part of the code needs attention I has several problems. Without the need to support snappy-ab most of it will go away including the rootPartition() and otherRootPartition() names. I look forward to that day :)